### ISSUE TITLE
Using oe_get_report_v2() to query whether the Enclave is running in debug mode causes SGX remote attestation related dependency checks

### ISSUE THREAD

**Issue by bodzhang** [openenclave/openenclave#3988]
The `oe_get_report_v2(flag=0,...) ` call, which does not specify any remote attestation related request, triggers remote attestation related flow. In an environment missing libsgx_quote_ex, the following error is encountered:

2021-04-23T16:32:15+0000.587941Z [(H)ERROR] tid(0x7fbc6363ac00) | _load_quote_ex_library_once() sgx_get_supported_att_key_id_num() status=16385 num=0
[/home/azureuser/mystikos/third_party/openenclave/openenclave/host/sgx/sgxquote.c:_load_quote_ex_library_once:287]
2021-04-23T16:32:15+0000.587959Z [(H)ERROR] tid(0x7fbc6363ac00) | :OE_QUOTE_PROVIDER_CALL_ERROR [/home/azureuser/mystikos/third_party/openenclave/openenclave/host/sgx/sgxquote.c:_load_quote_ex_library_once:288]

The ability to query whether the Enclave is running in debug mode, without triggering unnecessary remote attestation code, will be useful.

**Comment by yentsanglee** [comment]
Regardless of flag, oe_get_report_v2() initializes attesters which enumerates all built-in types including remote formats. (Search for `oe_sgx_get_supported_attester_format_ids`).

One possible fix is to ignore the missing libsgx_quote_ex error and come back with local attestation support only (search for `_get_attester_plugins`).

**Comment by dthaler** [comment]
Can we just deprecate this API in favor of `oe_get_evidence()`?

**Comment by radhikaj** [comment]
@bodzhang, do you want to determine if the enclave is in debug mode from inside the enclave or in the host?

**Comment by qiucwang** [comment]
Hi @bodzhang, missing libsgx_quote_ex will generate error messages but shouldn't terminate the `oe_get_report_v2()`. The attester_plugins will load `sgx_local_uuid` in any case. I just run and passed `ctest/report/test_local_report()` which calls  `oe_get_report_v2(flags=0)` with `SGX_AESM_ADDR` set and without `libsgx_quote_ex`. 
Could you share more context and complete logging? 
`sgx_get_qetarget_info()` will failed in this case though if you need it to get `target_info` for `oe_get_report_v2`.

**Comment by shnwc** [comment]
The real need is to get enclave debug mode, without triggering remote attestation code. Yen will check with Radhika.

**Comment by yentsanglee** [comment]
@radhikaj Is there any API to query if the enclave is running in debug mode? If no, should one be created?

**Comment by yentsanglee** [comment]
@bodzhang Is this still an issue or do you still need an API to query enclave mode? If you are unblocked and the API is no longer needed, let's close this issue.

**Comment by yentsanglee** [comment]
Closing the issue. Please re-activate if this is unresolved.
